### ISSUE TITLE
ZipImportError thrown when trying to build project

### ISSUE THREAD

**Issue by hahnicity** [rtfd/readthedocs.org#1105]
We've been trying to run setup.py install on our project but we've been getting a ZipImportError
```
install
-----

zip_safe flag not set; analyzing archive contents...
Traceback (most recent call last):
  File "setup.py", line 38, in <module>
    'development = dogwalker.development:main'
  File "/usr/lib/python2.7/distutils/core.py", line 152, in setup
    dist.run_commands()
  File "/usr/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/usr/lib/lldocs/readthedocs.org/user_builds/dogwalker/envs/latest/local/lib/python2.7/site-packages/setuptools/command/install.py", line 73, in run
    self.do_egg_install()
  File "/usr/lib/lldocs/readthedocs.org/user_builds/dogwalker/envs/latest/local/lib/python2.7/site-packages/setuptools/command/install.py", line 101, in do_egg_install
    cmd.run()
  File "/usr/lib/lldocs/readthedocs.org/user_builds/dogwalker/envs/latest/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 375, in run
    self.easy_install(spec, not self.no_deps)
  File "/usr/lib/lldocs/readthedocs.org/user_builds/dogwalker/envs/latest/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 587, in easy_install
    return self.install_item(None, spec, tmpdir, deps, True)
  File "/usr/lib/lldocs/readthedocs.org/user_builds/dogwalker/envs/latest/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 638, in install_item
    self.process_distribution(spec, dist, deps)
  File "/usr/lib/lldocs/readthedocs.org/user_builds/dogwalker/envs/latest/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 684, in process_distribution
    [requirement], self.local_index, self.easy_install
  File "/usr/lib/lldocs/readthedocs.org/user_builds/dogwalker/envs/latest/local/lib/python2.7/site-packages/pkg_resources.py", line 581, in resolve
    requirements.extend(dist.requires(req.extras)[::-1])
  File "/usr/lib/lldocs/readthedocs.org/user_builds/dogwalker/envs/latest/local/lib/python2.7/site-packages/pkg_resources.py", line 2323, in requires
    dm = self._dep_map
  File "/usr/lib/lldocs/readthedocs.org/user_builds/dogwalker/envs/latest/local/lib/python2.7/site-packages/pkg_resources.py", line 2308, in _dep_map
    for extra,reqs in split_sections(self._get_metadata(name)):
  File "/usr/lib/lldocs/readthedocs.org/user_builds/dogwalker/envs/latest/local/lib/python2.7/site-packages/pkg_resources.py", line 2750, in split_sections
    for line in yield_lines(s):
  File "/usr/lib/lldocs/readthedocs.org/user_builds/dogwalker/envs/latest/local/lib/python2.7/site-packages/pkg_resources.py", line 2009, in yield_lines
    for ss in strs:
  File "/usr/lib/lldocs/readthedocs.org/user_builds/dogwalker/envs/latest/local/lib/python2.7/site-packages/pkg_resources.py", line 2337, in _get_metadata
    for line in self.get_metadata_lines(name):
  File "/usr/lib/lldocs/readthedocs.org/user_builds/dogwalker/envs/latest/local/lib/python2.7/site-packages/pkg_resources.py", line 1325, in get_metadata_lines
    return yield_lines(self.get_metadata(name))
  File "/usr/lib/lldocs/readthedocs.org/user_builds/dogwalker/envs/latest/local/lib/python2.7/site-packages/pkg_resources.py", line 1317, in get_metadata
    return self._get(self._fn(self.egg_info,name))
  File "/usr/lib/lldocs/readthedocs.org/user_builds/dogwalker/envs/latest/local/lib/python2.7/site-packages/pkg_resources.py", line 1381, in _get
    return self.loader.get_data(path)
zipimport.ZipImportError: bad local file header in /usr/lib/lldocs/readthedocs.org/user_builds/dogwalker/envs/latest/lib/python2.7/site-packages/dogwalker-1.0-py2.7.egg
```

From a [quick google search](https://www.google.com/search?q=python+zipimporterror+bad+local+file+header+readthedocs&oq=python+zipimporterror+bad+local+file+header+readthedocs&aqs=chrome..69i57.677j0j7&sourceid=chrome&es_sm=93&ie=UTF-8) for this it seems like a lot of other projects have run into the same issue

**Comment by lmjohns3** [comment]
I was also seeing this on my project, but it does seem pretty intermittent.

I ran across this mention of a similar issue (from 2009!) in the same google search: http://code.activestate.com/lists/python-distutils-sig/10084/ -- not sure whether that's even still relevant though.

**Comment by gregmuellegger** [comment]
Does this happen in a build on readthedocs.org? I cannot find any dogwalker project there.

**Comment by lmjohns3** [comment]
FWIW I haven't seen this show up on any of my recent builds (my projects are theanets and downhill).

**Comment by gregmuellegger** [comment]
Awesome, so we might have fixed that already (maybe by accident ;)) I'm closing this for now, feel free to open up a new ticket if this occurs to you again.

**Comment by hoechenberger** [comment]
This problem occurs quite frequently for me for the automated builds after pushing to Github. If I then restart the build manually, it finishes just fine.

```
Adding pphelper 0.6 to easy-install.pth file

Installed /home/docs/checkouts/readthedocs.org/user_builds/pphelper/envs/doc/lib/python2.7/site-packages/pphelper-0.6-py2.7.egg
Processing dependencies for pphelper==0.6
Traceback (most recent call last):
  File "setup.py", line 43, in <module>
    'Programming Language :: Python :: 2.7']
  File "/usr/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/usr/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/usr/lib/python2.7/dist-packages/setuptools/command/install.py", line 73, in run
    self.do_egg_install()
  File "/usr/lib/python2.7/dist-packages/setuptools/command/install.py", line 96, in do_egg_install
    cmd.run()
  File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", line 381, in run
    self.easy_install(spec, not self.no_deps)
  File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", line 597, in easy_install
    return self.install_item(None, spec, tmpdir, deps, True)
  File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", line 648, in install_item
    self.process_distribution(spec, dist, deps)
  File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", line 694, in process_distribution
    [requirement], self.local_index, self.easy_install
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 633, in resolve
    requirements.extend(dist.requires(req.extras)[::-1])
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2291, in requires
    dm = self._dep_map
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2277, in _dep_map
    for extra,reqs in split_sections(self._get_metadata(name)):
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2715, in split_sections
    for line in yield_lines(s):
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 1989, in yield_lines
    for ss in strs:
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2305, in _get_metadata
    for line in self.get_metadata_lines(name):
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 1369, in get_metadata_lines
    return yield_lines(self.get_metadata(name))
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 1361, in get_metadata
    return self._get(self._fn(self.egg_info,name))
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 1425, in _get
    return self.loader.get_data(path)
zipimport.ZipImportError: bad local file header in /home/docs/checkouts/readthedocs.org/user_builds/pphelper/envs/doc/lib/python2.7/site-packages/pphelper-0.6-py2.7.egg
```

**Comment by gregmuellegger** [comment]
This looks exactly like the bug reported in https://bitbucket.org/pypa/setuptools/issues/168 - it was resolved in setuptools 3.5.2

We try to install setuptools 18.0.1 but are ending up with setuptools 3.3 as we can see in the output of the second command on this build: https://readthedocs.org/projects/pphelper/builds/3297879/

I assume that the setuptools installed system wide (via apt-get) takes precedence over the virtualenv one. So I guess the solution to this problem is to update the system wide setuptools.

@agjohnson Could you make this happen?

**Comment by ericholscher** [comment]
Believe this should be fixed, as we upgraded setuptools.
